Description: clean up CMakeLists.txt
 * Drop excess build dependencies that are not actually used.
 * Require KDE 4.4.0
 * Require Qt 4.6.0
 * Drop obsolete hacks for Plasma from <= KDE 4.1
Author: Modestas Vainius <modax@debian.org>
Forwarded: yes
Last-Update: 2010-05-08

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,21 +1,8 @@
 project(KTORRENT)
-find_package(KDE4 REQUIRED)
-find_package(KDE4Internal REQUIRED)
-find_package(GMP REQUIRED)
-find_package(QCA2 REQUIRED)
-find_package(Boost REQUIRED)
+find_package(Qt4 4.6.0 REQUIRED)
+find_package(KDE4 4.4.0 REQUIRED)
 find_package(KTorrent REQUIRED)
 
-# backward compatibility to <= KDE 4.1
-MACRO_ENSURE_VERSION( "4.1.83" ${KDE_VERSION} KDE_VERSION_OK)
-if(NOT KDE_VERSION_OK)
-  macro_optional_find_package(Plasma)
-  macro_log_feature(PLASMA_FOUND "Plasma" "Plasma libraries and header files" "http://plasma.kde.org/" FALSE "" "Plasma is needed for KTorrent Plasmoid")
-else(NOT KDE_VERSION_OK)
-  set(PLASMA_FOUND true)
-  set(PLASMA_LIBS ${KDE4_PLASMA_LIBS})
-endif(NOT KDE_VERSION_OK)
-
 # set some default settings
 include(KDE4Defaults)
 enable_testing()
@@ -53,9 +40,7 @@ add_subdirectory(plugins)
 add_subdirectory(ideal)
 add_subdirectory(ktorrent)
 add_subdirectory(ktupnptest)
-if( PLASMA_FOUND )
-    add_subdirectory(plasma)
-endif(PLASMA_FOUND)
+add_subdirectory(plasma)
 add_subdirectory(ktmagnetdownloader)
 
 # --- show log ---
--- a/plasma/applet/CMakeLists.txt
+++ b/plasma/applet/CMakeLists.txt
@@ -18,7 +18,7 @@ else(NOT QT_VERSION_OK)
 		kde4_add_ui_files(ktapplet_SRCS appletconfig.ui)
 		
 		kde4_add_plugin(plasma_applet_ktorrent ${ktapplet_SRCS})
-		target_link_libraries(plasma_applet_ktorrent  ${PLASMA_LIBS} ${KDE4_KIO_LIBS} ${TASKMANAGER_LIBRARY} ${LIBKTORRENT_LIBRARIES} ktcore)
+		target_link_libraries(plasma_applet_ktorrent  ${KDE4_PLASMA_LIBS} ${KDE4_KIO_LIBS} ${TASKMANAGER_LIBRARY} ${LIBKTORRENT_LIBRARIES} ktcore)
 		
 		install(TARGETS plasma_applet_ktorrent DESTINATION ${PLUGIN_INSTALL_DIR})
 		install(FILES plasma-applet-ktorrent.desktop DESTINATION ${SERVICES_INSTALL_DIR})
@@ -26,4 +26,4 @@ else(NOT QT_VERSION_OK)
 
 	macro_log_feature(TASKMANAGER_FOUND "libtaskmanager" "libtaskmanager library and header files" "http://www.kde.org/" FALSE "" "libtaskmanager is needed for KTorrent Plasmoid")
 
-endif(NOT QT_VERSION_OK)
\ No newline at end of file
+endif(NOT QT_VERSION_OK)
--- a/plasma/dataengine/CMakeLists.txt
+++ b/plasma/dataengine/CMakeLists.txt
@@ -3,7 +3,7 @@ set(ktengine_SRCS
 )
 
 kde4_add_plugin(plasma_engine_ktorrent ${ktengine_SRCS})
-target_link_libraries(plasma_engine_ktorrent ${KDE4_KDECORE_LIBS} ${PLASMA_LIBS} ${LIBKTORRENT_LIBRARIES})
+target_link_libraries(plasma_engine_ktorrent ${KDE4_KDECORE_LIBS} ${KDE4_PLASMA_LIBS} ${LIBKTORRENT_LIBRARIES})
 
 install(TARGETS plasma_engine_ktorrent DESTINATION ${PLUGIN_INSTALL_DIR})
 install(FILES plasma-dataengine-ktorrent.desktop DESTINATION ${SERVICES_INSTALL_DIR} )
